backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/utils/Modifier.kt

```diff
@@ -30,3 +33,12 @@ internal fun Modifier.hideContentIf(hide: Boolean): Modifier =
       }
     }
   }
+
+
+@OptIn(ExperimentalContracts::class)
+internal inline fun Modifier.applyIf(predicate: Boolean, block: Modifier.() -> Modifier): Modifier {
+  contract {
```

Comment:
I still left the contract here in case the compiler learns how to understand it in the future.

Reply:
The contract just specifies that it is only called once in place, not that the result is not nullable, so this will never work - you need a different contract/function for this

Reply:
It doesn't work via results. A compiler can understand that the lambda will be called in place and therefore it can analyse that all the smart checks are done before the lambda will be valid. But there's no way to state that a predicate will be true in this case as well for the compiler.

Reply:
this might explain what i mean better: https://www.baeldung.com/kotlin/contracts you need a contract like this ` returns() implies (request != null)` 
the call in place only verifies that it'll be called once and can be used for initialising things - it does not do any further smart checks